functions/src/modules/SQR.functions.ts

```diff
@@ -214,65 +215,71 @@ export const processSubmissions = functions.database.ref('/webforms/sqr/{submiss
-
     let templateNode = await db.ref('/email/templates').orderByKey()
-    .equalTo(fileName).once('value');
-
+                                .equalTo(fileName).once('value');
     let template = templateNode.val();
 
     if (template.exists()) {
-        await helpers.updateTemplate(template.id, htmlContent);
-        /////////////////////////////////////////////////////////////////
+        
+        const tempLocalFile = path.join(os.tmpdir(), fileName);
+        let emailRef = bucket.file(filePath);    
+        await emailRef.download({ destination: tempLocalFile });
+
+        // Use the following to upate the template on SendInBlue
+        let htmlContent = fs.readFileSync(tempLocalFile, 'utf-8');
+        let { newTemplate, sender, bcc, subject, sample } = template;
+        let id;
+
+        if (newTemplate) {
+            id = await emailHelper.createTemplate(fileName, htmlContent, subject);
+            await db.ref(`/email/templates/${fileName}`).push({
+                id,
+                lastUpdated: new Date(),
+                version: 1,
+                sample
+            });
+        }            
+        else {
+            id = await emailHelper.getTemplateId(fileName);
+            await emailHelper.updateTemplate(id, htmlContent);
+            await db.ref(`/email/templates/${template.key}`)
+                .update({ lastUpdated: new Date() });
+        }
+
         // 2. Send a test Email confirming it has been updated correctly
-        /////////////////////////////////////////////////////////////////
-        let { params, config } = template.sample;
-        let { id } = template;
-        await helpers.sendEmail(config.to, config.bcc, id, params);
-    } 
-    else {
-        let subject = 'WHERE CAN I GET subject FROM'; // mandatory
-        let id = await helpers.createTemplate(fileName, htmlContent, subject);
-        await db.ref(`/email/templates/${fileName}`).push({
-            id,
-            lastUpdated: new Date(),
-            version: 1
-            // Need some way to fill `sample`
+        db.ref(`/email/notifications`).push({
+            template: fileName,
+            to: sender.email,
+            bcc,
+            params: sample.params
         });
-        /////////////////////////////////////////////////////////////////
-        // 2. Send a test Email confirming it has been updated correctly
-        /////////////////////////////////////////////////////////////////
-        // WAITING TO KNOW SOURCE OF params`, `config`        
-    }    
+    } else 
+        console.log("Template metadata doesn't exist.");
+}
+
+export const updateTemplatesOnTemplateUpload = functions.storage.object()
+.onFinalize(object => { // called when either a NEW object is created, or when an object is overwritten
+    updateEmailTemplates(object.name);
     return 1;
 });
 
+export const updateTemplatesOnMetadataChange = functions.database.ref('/email/templates}')
+.onUpdate(async (change, context) => {    
```

Comment:
Why only `onUpdate`? Or it is also fired on creating a node?

Reply:
I don't think we should trigger the function on **creating** a new node as the template wouldn't exist yet at the storage... 
A node is created and then on uploading the template that matches the `template` field in this node the function is automatically triggered.